src/maths.rs

```diff
@@ -330,11 +342,17 @@ impl MathematicalOps for Decimal {
         (Decimal::ONE + (self / Decimal::from_parts(2318911239, 3292722, 0, false, 16)).erf()) / TWO
     }
 
-    /// The Probability density function for a Normal distribution
+    /// The Probability density function for a Normal distribution.
     fn norm_pdf(&self) -> Decimal {
         let sqrt2pi = Decimal::from_parts_raw(2133383024, 2079885984, 1358845910, 1835008);
         (-self.powi(2) / TWO).exp() / sqrt2pi
     }
+
+    /// The Probability density function for a Normal distribution, return `None` on overflow.
+    fn checked_norm_pdf(&self) -> Option<Decimal> {
+        let sqrt2pi = Decimal::from_parts_raw(2133383024, 2079885984, 1358845910, 1835008);
+        Some((-self.powi(2) / TWO).checked_exp()? / sqrt2pi)
```

Comment:
I'd almost be inclined to share the logic with `norm_pdf` too as opposed to have two copies. The other `checked` function which should be called is `checked_powi` since it is available, as well as `checked_div`. They're a little overkill however then cover all compiler driven possible cases.

Reply:
I think it would be easy to pass function pointers into an internal method.